test/jdk/java/awt/image/VolatileImage/ReportRenderingError.java

```diff
@@ -0,0 +1,132 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Color;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.GraphicsEnvironment;
+import java.awt.Image;
+import java.awt.image.BufferedImage;
+import java.awt.image.ImageObserver;
+import java.awt.image.ImageProducer;
+
+/*
+ * @test id=default
+ * @bug 8272288
+ * @key headful
+ * @summary Broken rendering should be reported by the contentsLost()
+ *
+ * @run main/othervm ReportRenderingError
+ */
+
+/*
+ * @test id=windows
+ * @bug 8272288
+ * @key headful
+ * @summary Broken rendering should be reported by the contentsLost()
+ * @requires (os.family == "windows")
+ *
+ * @run main/othervm -Dsun.java2d.opengl=True ReportRenderingError
+ * @run main/othervm -Dsun.java2d.d3d=True    ReportRenderingError
+ * @run main/othervm -Dsun.java2d.d3d=false   ReportRenderingError
+ */
+
+/*
+ * @test id=macos
+ * @bug 8272288
+ * @key headful
+ * @summary Broken rendering should be reported by the contentsLost()
+ *
+ * @requires (os.family == "mac")
+ * @run main/othervm -Dsun.java2d.opengl=True ReportRenderingError
+ * @run main/othervm -Dsun.java2d.metal=True  ReportRenderingError
+ */
+
+/*
+ * @test id=linux
+ * @bug 8272288
+ * @key headful
+ * @summary Broken rendering should be reported by the contentsLost()
+ *
+ * @requires (os.family == "linux")
+ * @run main/othervm -Dsun.java2d.opengl=True   ReportRenderingError
+ * @run main/othervm -Dsun.java2d.xrender=True  ReportRenderingError
+ * @run main/othervm -Dsun.java2d.xrender=false ReportRenderingError
+ */
+public final class ReportRenderingError {
+
+    public static void main(String[] args) {
+        var gc = GraphicsEnvironment.getLocalGraphicsEnvironment()
+                                    .getDefaultScreenDevice()
+                                    .getDefaultConfiguration();
+        var vi = gc.createCompatibleVolatileImage(10, 10);
+
+        Image image = new EmptyImage();
+        BufferedImage snapshot;
+        int attempt = 0;
+        do {
+            vi.validate(gc);
+            Graphics2D g = vi.createGraphics();
+            g.setColor(Color.RED);
+            g.drawImage(image, 0, 0, null); // <- can cause InvalidPipeException
```

Comment:
This drawImage call is expected to through InvalidPipeException all the times or is it intermittent?
If it is intermittent why so? Please explain. It looks like after https://github.com/openjdk/jdk/commit/8da6c8d6 we will through InvalidPipeException for all cases of custom images.

Reply:
It is not really intermittent but depends on the java2d pipeline currently in use. InvalidPipeException is produced by the accelerated pipelines but code works fine on GDI for example.

Reply:
Thanks for the clarification